doc/perception_pipeline/src/cylinder_segment.cpp

```diff
@@ -0,0 +1,314 @@
+  Eigen::Vector3d origin_z_direction(0., 0., 1.);
+  Eigen::Vector3d axis;
+  double angle;
+  axis = origin_z_direction.cross(cylinder_z_direction);
+  axis.normalize();
+  angle = acos(cylinder_z_direction.dot(origin_z_direction));
+  cylinder_pose.orientation.x = axis.x() * sin(angle / 2);
+  cylinder_pose.orientation.y = axis.y() * sin(angle / 2);
+  cylinder_pose.orientation.z = axis.z() * sin(angle / 2);
+  cylinder_pose.orientation.w = cos(angle / 2);
+
+  // Setting the position of cylinder.
+  cylinder_pose.position.x = cylinder_params->center_pt[0];
+  cylinder_pose.position.y = cylinder_params->center_pt[1];
+  cylinder_pose.position.z = cylinder_params->center_pt[2];
+
+  // Add cylinder as collision object
+  collision_object.primitives.push_back(primitive);
+  collision_object.primitive_poses.push_back(cylinder_pose);
+  collision_object.operation = collision_object.ADD;
+  std::vector<moveit_msgs::CollisionObject> collision_objects;
+  collision_objects.push_back(collision_object);
+  planning_scene_interface.applyCollisionObjects(collision_objects);
+  // END_SUB_TUTORIAL
+}
+
+/** \brief Given the pointcloud containing just the cylinder, compute its center point and its height and store in cylinder_params.
+    @param cloud - Pointcloud containning just the cylinder. 
+    @param cylinder_params - Pointer to the struct add_cylinder_params. */
+void extractLocationHeight(pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloud, add_cylinder_params* cylinder_params)
+{
+  double max_angle_y = 0.0;
+  double min_angle_y = std::numeric_limits<double>::infinity();
+
+  double lowest_point[3];
+  double highest_point[3];
+  // BEGIN_SUB_TUTORIAL extract_location_height
+  // Consider a point inside the point cloud and imagine that point is formed on a XY plane where the perpendicular
+  // distance from the plane to the camera is Z. |br|
+  // The perpendicular drawn from the camera to the plane hits at center of the XY plane. |br|
+  // We have the x and y coordinate of the point which is formed on the XY plane. |br|
+  // X is the horizontal axis and Y is the vertical axis. |br|
+  // C is the center of the plane which is Z meter away from the center of camera and A is any point on the plane. |br|
+  // Now we know Z is the perpendicular distance from the point to the camera. |br|
+  // If you need to find the  actual distance d from the point to the camera, you should calculate the hypotenuse-
+  // |code_start| hypot(pt.z, pt.x);\ |code_end| |br|
+  // angle the point made horizontally- |code_start| atan2(pt.z,pt.x);\ |code_end| |br|
+  // angle the point made Verticlly- |code_start| atan2(pt.z, pt.y);\ |code_end| |br|
+  // Loop over the entire pointcloud using BOOST_FOREACH to have a simpler nicer looking implimentation.
+  BOOST_FOREACH (const pcl::PointXYZRGB& pt, cloud->points)
+  {
+    // Find the coordinates of the highest point
```

Comment:
These should be /**/ comments so that they don't break up the auto-generated tutorial